airbyte-webapp/src/views/Connector/ConnectorForm/components/Property/SecretConfirmationControl.tsx

```diff
@@ -0,0 +1,114 @@
+import { useField, useFormikContext } from "formik";
+import React, { useEffect, useRef, useState } from "react";
+import { FormattedMessage } from "react-intl";
+import styled from "styled-components";
+
+import { Button } from "components/ui/Button";
+import { Input } from "components/ui/Input";
+import { SecretTextArea } from "components/ui/SecretTextArea";
+
+const ComponentContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+`;
+
+interface SecretConfirmationControlProps {
+  showButtons?: boolean;
+  name: string;
+  multiline: boolean;
+  disabled?: boolean;
+  error?: boolean;
+}
+
+const SecretConfirmationControl: React.FC<SecretConfirmationControlProps> = ({
+  showButtons,
+  disabled,
+  multiline,
+  name,
+  error,
+}) => {
+  const [field, , helpers] = useField(name);
+  const [previousValue, setPreviousValue] = useState<unknown>(undefined);
+  const isEditInProgress = Boolean(previousValue);
+  const controlRef = useRef<HTMLElement>(null);
+
+  const { dirty, touched } = useFormikContext();
+
+  const component =
+    multiline && (isEditInProgress || !showButtons) ? (
+      <SecretTextArea
+        {...field}
+        autoComplete="off"
+        value={field.value ?? ""}
+        rows={3}
+        disabled={disabled}
+        error={error}
+      />
+    ) : (
+      <Input
+        {...field}
+        autoComplete="off"
+        value={field.value ?? ""}
+        type="password"
+        disabled={disabled}
+        error={error}
+      />
+    );
+
+  useEffect(() => {
+    if (!dirty && !touched && previousValue) {
+      setPreviousValue(undefined);
+    }
+  }, [dirty, helpers, previousValue, touched]);
+
+  if (!showButtons) {
+    return <>{component}</>;
+  }
+
+  const handleStartEdit = () => {
+    if (controlRef && controlRef.current) {
+      controlRef.current?.removeAttribute?.("disabled");
+      controlRef.current?.focus?.();
+    }
+    setPreviousValue(field.value);
+    helpers.setValue("");
+  };
+
+  const onDone = () => {
+    setPreviousValue(undefined);
+  };
+
+  const onCancel = () => {
+    if (previousValue) {
+      helpers.setValue(previousValue);
+    }
+    setPreviousValue(undefined);
+  };
+
+  return (
+    <ComponentContainer>
+      {React.cloneElement(component, {
```

Comment:
Why do we need `cloneElement` here? Could we not just set the ref and the other attributes here to those values directly on the`<SecretTextArea>` and `<Input>` above?

Reply:
good point, we don't - simplified this.